linkerd/app/inbound/src/direct.rs

```diff
@@ -61,16 +61,17 @@ impl<T> Inbound<T> {
     /// 2. TLS is required;
     /// 3. A transport header is expected. It's not strictly required, as
     ///    gateways may need to accept HTTP requests from older proxy versions
-    pub fn push_direct<I, TSvc, G, GSvc>(
+    pub fn push_direct<A, I, TSvc, G, GSvc>(
```

Comment:
nit: let's use `T` and change `T/TSvc` to `N/NSvc`. I've found being consistent about this helps a lot

Reply:
Done!